Purchases/Networking/Backend.swift

```diff
@@ -237,20 +241,14 @@ class Backend {
                 let signature = signatureData["signature"] as? String
                 let keyIdentifier = offer["key_id"] as? String
                 let nonceString = signatureData["nonce"] as? String
-                let maybeNonce: UUID?
-                if let nonceString = nonceString {
-                    maybeNonce = UUID(uuidString: nonceString)
-                } else {
-                    maybeNonce = nil
-                }
-
-                let timestamp = signatureData["timestamp"] as? Int
+                let maybeNonce = nonceString.flatMap { UUID(uuidString: $0) }
 
-                completion(signature, keyIdentifier, maybeNonce, timestamp as NSNumber?, nil)
```

Comment:
I think I prefer the older version without the cast

Reply:
The old version has a cast `timestamp as NSNumber?` because the completion block has `NSNumber`. Here I'm removing the cast because I updated the completion block to an `Int`

Reply:
What I did was move the cast to where it is required: Only for the StoreKit API. The idea being: don't use `NSNumber` until we absolutely have to.

Reply:
Let me clarify. The previous version was doing this already:

```
let timestamp = signatureData["timestamp"] as? Int
```

So I am basically saying to leave it like this:

```
let timestamp = signatureData["timestamp"] as? Int
completion(signature, keyIdentifier, maybeNonce, timestamp, nil)
```

Reply:
🤦‍♂️ got it.